viashpy/_run.py

```diff
@@ -79,8 +79,9 @@ def viash_run(
     if not config.is_file():
         raise FileNotFoundError(f"{config} does not exist or is not a file.")
     return check_output(
-        [viash_location, "run", config, "--"]
-        + _add_cpu_and_memory(args, cpus, memory, "--"),
+        _add_cpu_and_memory([viash_location, "run", config], cpus, memory, "--")
+        + ["--"]
+        + args,
```

Comment:
FYI; I think

```suggestion
        [viash_location, "run", config, "--"]
        + _add_cpu_and_memory(args, cpus, memory, "---"),
```

Would also have worked.

Reply:
Also, why pass any args at all? Why not just

```suggestion
        [viash_location, "run", config, "--"]
        + _add_cpu_and_memory([], cpus, memory, "---")
        + args

Reply:
Turns out I forgot that `---memory` can also be used with `viash run`. `viash run` builds an executable and everything after `--` gets passed to it (including `---memory` and `---cpus`). Good suggestion!